generators/app/templates/gulp/tasks/templates.js

```diff
@@ -1,12 +1,31 @@
 'use strict';
 
 var fs = require('fs');
+var path = require('path');
 
 var templatesTask = function (gulp, plugins, config, helpers) {
 
+  function getTemplateInputData(file) {
+    try {
+      var data = JSON.parse(
+        fs.readFileSync(config.src.dataPath + path.basename(file.path) + '.json')
+      );
+    } catch(e) {
```

Comment:
I believe @jakub300 was joking... Obviously doing nothing is faster than anything. 😉 
What do you mean by the order? At the time the pipe reaches this function, the template should already be loaded into memory or at least have an open file handle...
Also this tries to read the file, so there's no use for checking for directory and it shouldn't affect performance in a visible way.